utils/cSoundPlayer.cpp

```diff
@@ -5,91 +5,170 @@
  *      Author: Stefan
  */
 
-#include "../include/d2tmh.h"
+#include "cSoundPlayer.h"
 
-cSoundPlayer::cSoundPlayer(int maxVoices) {
-	maximumVoices = maxVoices;
-	memset(voices, -1, sizeof(voices));
-}
+#include "allegroh.h"
+#include "definitions.h"
+#include "utils/cLog.h"
 
-cSoundPlayer::~cSoundPlayer() {
+#include <fmt/core.h>
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+
+namespace {
+
+constexpr int kAllegroMaxNrVoices = 256; // From the Allegro 4 documentation
+constexpr int kAllegroMaxVolume = 255;
+constexpr int kAllegroPanCenter = 128;
+
+constexpr int kFinishedPLaying = -1;
+constexpr int kNoVoiceAvailable = -1;
+constexpr int kNoVoice = -1;
+constexpr int kNoLoop = 0;
+
+constexpr int kMinNrVoices = 4;
+constexpr int kMaxVolume = 220;
```

Comment:
This is interesting. Now that more audio stuff is together, we can see there are two different max volumes in the code. 
I think this one can be removed, as all samples are played with either a hardocded volume or the Allegro maximum.
It's only used to set the music volume to 110. (See a FIXME below). I think we can replace this with a max music volume of 110 (or 128?).

Reply:
I am unsure about this.  I noticed the max volume returns an Allegro max value, but the intention is to return the max intended volume at some point. So perhaps I made a distinction somewhere sometime for that. 

Regardless, it would make sense to have only one "max" volume  and using the Allegro one would make sense then.